### ISSUE TITLE
CLI 1.35.4-RC1 crashing with a --cddl-format error

### ISSUE THREAD

**Issue by gitmachtl** [input-output-hk/cardano-node#4597]
``` console
$ cardano-cli version
cardano-cli 1.35.4 - linux-x86_64 - ghc-8.10
git rev 54998e843cf7d54eb1594ce191a261a112429ead
```

Using the `--cddl-format` flag with this revision in the `transaction build-raw` command causes an error:
`Invalid option '--cddl-format' `

Also the two lines
```
--cddl-format            Serialise in the ledger CDDL specified CBOR format.
 --cli-format            Serialise in the cardano-cli CBOR format.
```
are missing when checking the syntax with this revision like
``` console
cardano-cli transaction build-raw
```

if cddl-format is now the default and only format, the cli should still accept the flag for a while to change the calling in 3rd party tools.

**Comment by ada4profit** [comment]
It is not a bug, but many 3rd party tooling are depending on this parameter that is removed and as such it is breaking their solution. Basically it should have skipped this parameter if provided by the 3rd party solution, instead of giving an error message. This would have been a much more elegant solution dealing with this issue. Now 3rd parties have to go over their code and remove it on basis of something they were not informed about. Jmho.

**Comment by Jimbo4350** [comment]
> It is not a bug, but many 3rd party tooling are depending on this parameter that is removed and as such it is breaking their solution. Basically it should have skipped this parameter if provided by the 3rd party solution, instead of giving an error message. This would have been a much more elegant solution dealing with this issue. Now 3rd parties have to go over their code and remove it on basis of something they were not informed about. Jmho.

Ok, we can adjust it so that it does not fail.

**Comment by gitmachtl** [comment]
thx @Jimbo4350, if the "dummy --cddl-format" flag is in for the next releases that would be perfect. so 3rd party tools have time to set a new node/cli min version in there tools and also remove the --cddl-flag from the calling commands.

**Comment by gitmachtl** [comment]
Thx @newhoggy, #4617 looks good. The `transaction build/build-raw` command is now working again without an error.
We will update the tools to remove the `--cddl-format` calling flag once 1.35.4 is released so we can set the minCliVersion to 1.35.4. Ready for a full removal starting with `1.36.*`
